src/social_network_link_prediction/dimensionality_reduction_methods/svd.py

```diff
@@ -0,0 +1,65 @@
+import networkx as nx
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy.sparse import csr_matrix, spdiags
+from scipy.sparse.linalg import svds
+
+def link_prediction_svd(G: nx.Graph, k = 5, normalize=False) -> csr_matrix:
+
+    # Create the adjacency matrix of the graph
+    adj_matrix = nx.adjacency_matrix(G)
+    
+    # Perform the singular value decomposition
+    U, S, Vt = svds(csr_matrix(adj_matrix).astype(float), k=k)
+    
+    # Make the diagonal matrix sparse
+    S = spdiags(S, [0], k, k)
+
+    # Compute the predicted adj_matrix
+    predicted_adj_matrix = ((U @ S) @ Vt)
+
+    if normalize:
+        # Normalize the predicted edge weights to the range [0, 1]
+        rows, cols = np.nonzero(predicted_adj_matrix)
+        max_weight = np.max(predicted_adj_matrix[rows, cols])
+        for i in range(len(rows)):
+            predicted_adj_matrix[rows[i], cols[i]] /= max_weight
```

Comment:
Potrebbe essere fatto con la funzione `zip`, magari è più veloce ??

```suggestion
        for row, col in zip(row, cols):
            predicted_adj_matrix[row, col] /= max_weight
```